app/script/entity/message/MemberMessage.js

```diff
@@ -42,6 +60,21 @@ z.entity.MemberMessage = class MemberMessage extends z.entity.SystemMessage {
         .map(userEntity => userEntity);
     });
 
+    this.joinedUserEntities.subscribe(joinedUserEntities => {
+      const visibleUsers = joinedUserEntities.slice();
+      this.exceedsMaxVisibleUsers(visibleUsers.length > MemberMessage.CONFIG.MAX_USERS_VISIBLE);
+      if (this.exceedsMaxVisibleUsers()) {
+        visibleUsers.splice(MemberMessage.CONFIG.REDUCED_USERS_COUNT);
+        const selfUser = visibleUsers.find(userEntity => userEntity.is_me);
+        if (selfUser) {
+          visibleUsers.pop();
+          visibleUsers.push(selfUser);
```

Comment:
The logic seems off. We need to make sure that the self user is never added twice to the visible users but always added as the last enttiy.